frontend/packages/kubevirt-plugin/src/components/vm-snapshots/vm-snapshots.tsx

```diff
@@ -81,17 +93,46 @@ export const VMSnapshotsPage: React.FC<VMLikeEntityTabProps> = ({ obj: vmLikeEnt
   const vmName = getName(vmLikeEntity);
   const namespace = getNamespace(vmLikeEntity);
 
-  const resource: WatchK8sResource = React.useMemo(
-    () => ({
-      isList: true,
-      kind: VirtualMachineSnapshotModel.kind,
-      namespaced: true,
-      namespace,
-    }),
-    [namespace],
+  const snapshotResource: WatchK8sResource = {
+    isList: true,
+    kind: VirtualMachineSnapshotModel.kind,
+    namespaced: true,
+    namespace,
+  };
+
+  const restoreResource: WatchK8sResource = {
+    isList: true,
+    kind: VirtualMachineRestoreModel.kind,
+    namespaced: true,
+    namespace,
+  };
+
+  const [snapshots, snapshotsLoaded, snapshotsError] = useK8sWatchResource<VMSnapshot[]>(
+    snapshotResource,
+  );
+  const [restores, restoresLoaded, restoresError] = useK8sWatchResource<VMRestore[]>(
+    restoreResource,
+  );
+
+  const mappedRelevantRestores = React.useMemo(
+    () =>
+      restores.reduce((restoreMap, currentRestore) => {
```

Comment:
this would be simpler

```tsx
      restores.reduce((restoreMap, currentRestore) => {
        const relevantRestore = restoreMap[getVmRestoreSnapshotName(currentRestore)];
        if (!relevantRestore || new Date(getVmRestoreTime(relevantRestore)).getTime() < new Date(getVmRestoreTime(currentRestore)).getTime()) {
            restoreMap[getVmRestoreSnapshotName(currentRestore)] = currentRestore
        }
        return restoreMap
      }, {}),
```